File: RPICommunicator_V1/app/src/main/java/com/example/rpicommunicator_v1/Database/BikeTourDao.java
package com.example.rpicommunicator_v1.Database;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;



    @Dao
    public interface BikeTourDao {

        @Insert
        void insert(BikeTour plant);

        @Update
        void update(BikeTour plant);

        @Delete
        void delete(BikeTour plant);

        @Query("SELECT * FROM biketour_table")
        LiveData<List<BikeTour>> getAllTours();


    }

File: app/src/main/java/com/example/rpicommunicator_v1/Database/BikeTourDao.kt
package com.example.rpicommunicator_v1.Database

import com.example.rpicommunicator_v1.Database.BikeTour
import androidx.lifecycle.LiveData
import androidx.room.*

@Dao
interface BikeTourDao {
    @Insert
    fun insert(plant: BikeTour?)

    @Update
    fun update(plant: BikeTour?)

    @Delete
    fun delete(plant: BikeTour?)

    @get:Query("SELECT * FROM biketour_table")
    val allTours: LiveData<List<BikeTour>>
}